egs/aspire/s5/local/chain/tuning/run_tdnn_lstm_1a.sh

```diff
@@ -0,0 +1,317 @@
+    --egs.chunk-width 160,140,110,80 \
+    --egs.chunk-left-context $chunk_left_context \
+    --egs.chunk-right-context $chunk_right_context \
+    --egs.chunk-left-context-initial 0 \
+    --egs.chunk-right-context-final 0 \
+    --egs.dir "$common_egs_dir" \
+    --cleanup.remove-egs $remove_egs \
+    --feat-dir $train_data_dir \
+    --tree-dir $treedir \
+    --lat-dir $lat_dir \
+    --dir $dir  || exit 1;
+fi
+
+graph_dir=$dir/graph_pp
+if [ $stage -le 14 ]; then
+  # Note: it might appear that this $lang directory is mismatched, and it is as
+  # far as the 'topo' is concerned, but this script doesn't read the 'topo' from
+  # the lang directory.
+  utils/mkgraph.sh --self-loop-scale 1.0 data/lang_pp_test $dir $graph_dir
+fi
+
+if [ $stage -le 15 ]; then
+  rm $dir/.error 2>/dev/null || true
+
+  for d in dev_rvb test_rvb; do
+    (
+      if [ ! -f exp/nnet3${nnet3_affix}/ivectors_${d}/ivector_online.scp ]; then
+        steps/online/nnet2/extract_ivectors_online.sh --cmd "$train_cmd" --nj 30 \
+          data/${d}_hires exp/nnet3${nnet3_affix}/extractor \
+          exp/nnet3${nnet3_affix}/ivectors_${d} || { echo "Failed i-vector extraction for data/${d}_hires"; touch $dir/.error; }
+      fi
+
+      decode_dir=$dir/decode_${d}_pp
+      steps/nnet3/decode.sh --nj 30 --cmd "$decode_cmd" --config conf/decode.config \
+        --acwt 1.0 --post-decode-acwt 10.0 \
+        --extra-left-context $extra_left_context \
+        --extra-right-context $extra_right_context \
+        --extra-left-context-initial 0 --extra-right-context-final 0 \
+        --frames-per-chunk 160 \
+        --online-ivector-dir exp/nnet3${nnet3_affix}/ivectors_${d} \
+        $graph_dir data/${d}_hires $decode_dir || { echo "Failed decoding in $decode_dir"; touch $dir/.error; }
+    ) &
+  done
+  wait
+
+  if [ -f $dir/.error ]; then
+    echo "Failed decoding."
+    exit 1
+  fi
+fi
+
+if [ $stage -le 16 ]; then
+  # %WER 22.9 | 2083 25838 | 81.6 12.0 6.4 4.6 22.9 70.8 | -0.469 | exp/chain/tdnn_lstm_1a/decode_dev_aspire_uniformsegmented_win10_over5_v8_iterfinal_pp_fg/score_8/penalty_0.0/ctm.filt.filt.sys
+  local/nnet3/prep_test_aspire.sh --stage $test_stage --decode-num-jobs 30 --affix "$affix" \
+   --acwt 1.0 --post-decode-acwt 10.0 \
+   --window 10 --overlap 5 \
+    --extra-left-context $extra_left_context \
+    --extra-right-context $extra_right_context \
+    --extra-left-context-initial 0 --extra-right-context-final 0 \
+   --sub-speaker-frames 6000 --max-count 75 --ivector-scale 0.75 \
+   --pass2-decode-opts "--min-active 1000" \
+   dev_aspire data/lang $dir/graph_pp $dir
+fi
+
+if [ $stage -le 17 ]; then
+  # %WER 24.0 | 2083 25822 | 79.9 11.9 8.1 3.9 24.0 71.7 | -0.450 | exp/chain/tdnn_lstm_1a_online/decode_dev_aspire_uniformsegmented_win10_over5_v8_iterfinal_pp_fg/score_10/penalty_0.0/ctm.filt.filt.sys
+  local/nnet3/prep_test_aspire_online.sh --stage $test_stage --decode-num-jobs 30 --affix "$affix" \
```

Comment:
wasn't one of the things I was asking for to rename these prep_test scripts to decode?
And I thought you were going to get rid of the excessively decorated directory names?

Reply:
I removed some of the affixes, especially in the training script. The decode has a lot of affixes, because the uniform segmentation approach uses a two pass decoding.